src/util/createInstantSearch.js

```diff
@@ -0,0 +1,108 @@
+import instantsearch from 'instantsearch.js/es';
+import algoliaHelper from 'algoliasearch-helper';
+const { SearchParameters, SearchResults } = algoliaHelper;
+import { _objectSpread } from './polyfills';
+import { warn } from './warn';
+
+export const createInstantSearch = instantSearchOptions => {
+  const search = instantsearch(instantSearchOptions);
+  const { searchClient, indexName } = instantSearchOptions;
+
+  // main API for SSR, called in asyncData of a root component which contains instantsearch
+  search.findResultsState = params => {
+    search.helper = algoliaHelper(
+      searchClient,
+      indexName,
+      _objectSpread({}, params, {
+        // parameters set by default
+        highlightPreTag: '__ais-highlight__',
+        highlightPostTag: '__/ais-highlight__',
+      })
+    );
+
+    return search.helper.searchOnce().then(({ content: lastResults }) => {
+      // The search instance needs to act as if this was a regular `search`
+      // but return a promise, since that is the interface of `asyncData`
+      search.helper.lastResults = lastResults;
+    });
+  };
+
+  // make sure correct data is available in each widget's state
+  // called in widget mixin
+  search.__forceRender = widget => {
+    if (!search.helper) {
+      warn(
+        'You did not call `instantsearch.findResultsState`, which is required for ais-instant-search-ssr'
```

Comment:
Try not addressing directly to the user, it's almost insulting haha

```suggestion
        '`instantsearch.findResultsState()` needs to be called when using `ais-instant-search-ssr`.'
```